olive/data/component/pre_process_data.py

```diff
@@ -122,7 +122,119 @@ def _tokenizer_and_align_labels(examples):
         tokenized_inputs["label"] = new_labels
         return tokenized_inputs
 
-    tokenized_datasets = _huggingface_pre_precess_helper(
+    tokenized_datasets = _huggingface_pre_process_helper(
         _dataset, model_name, input_cols, label_cols, _tokenizer_and_align_labels, **kwargs
     )
     return BaseDataset(tokenized_datasets, label_cols=["label"], max_samples=max_samples)
+
+
+@Registry.register_pre_process()
+def text_generation_huggingface_pre_process(
+    _dataset, model_name, input_cols, seqlen, stride=None, max_samples=None, **kwargs
+):
+    """Pre-process data.
+
+    Args:
+        _dataset (object): Data to be pre-processed.
+        model_name (str): Name of the huggingface model.
+        input_cols (list): List of input columns.
+        seqlen (int): Length of the sequence.
```

Comment:
Better to make these several `numbers` clearer.

Reply:
Do you mean better descriptions for the parameters? I can do that

Reply:
Sure! I will update these. Thanks!